File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/FrontEnd/src/app/Module/MReceiptNote/ReceiptNote.Entity.ts
import { IEntity } from "../IEntity.Entity";
import { InventoryEntity } from "../MInventory/Inventory.Entity";
import { SupplierEntity } from "../MSupplier/Supplier.Entity";
import { WareHouseEntity } from "../MWareHouse/WareHouse.Entity";
import { ProductEntity } from "../MProduct/Product.Entity";
import { EmployeeEntity } from "../MEmployee/Employee.Entity";
import { ReceiptNoteLineEntity } from "../MReceiptNoteLine/ReceiptNoteLine.Entity";

import { Data } from "@angular/router";

export class ReceiptNoteEntity extends IEntity {
    Id: number = 0;
    SupplierId: string = null;
    WareHouseId: string = null;
    EmployeeId: string = null;
    InvoiceDate: Date = null;
    Lock: boolean = null;
    Title: string = null;
    CreatedDate: Date = null;
    UpdatedDate: Date = null;
    ReceiptNoteNo: number = null;
    ReceiptDate: Date = null;
    SystemDate: Date = null;
    Address: string = null;
    PhoneNumber: string = null;
    Comment: string = null;
    Cx: number = null;
    SupplierEntity: SupplierEntity = new SupplierEntity();
    WareHouseEntity: WareHouseEntity = new WareHouseEntity();
    EmployeeEntity: EmployeeEntity = new EmployeeEntity();
    ReceiptNoteLineEntities: Array<ReceiptNoteLineEntity> = [];
    public constructor() {
        super();
    }
}

File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/Entities/ReceiptNoteEntity.cs
using EShop.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EShop.Entities
{
    public class ReceiptNoteEntity : BaseEntity
    {
        public Guid Id { get; set; }
        public Int32 ReceiptNoteNo { get; set; }
        public DateTime InvoiceDate { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime UpdatedDate { get; set; }
        public Boolean Lock { get; set; }
        public String Title { get; set; }
        public DateTime? ReceiptDate { get; set; }
        public DateTime? SystemDate { get; set; }
        public String Address { get; set; }
        public String PhoneNumber { get; set; }
        public String Comment { get; set; }
        public Guid WareHouseId { get; set; }
        public Guid SupplierId { get; set; }
        public Guid EmployeeId { get; set; }
        public EmployeeEntity EmployeeEntity { get; set; }
        public SupplierEntity SupplierEntity { get; set; }
        public WareHouseEntity WareHouseEntity { get; set; }
        public List<ReceiptNoteLineEntity> ReceiptNoteLineEntities { get; set; }

        public ReceiptNoteEntity():base() { }

        public ReceiptNoteEntity(ReceiptNote ReceiptNote, params object[] args) :base(ReceiptNote)
        {
		    foreach(object arg in args)
			{
                if (arg is Employee Employee)
                    EmployeeEntity = new EmployeeEntity(Employee);				
                if (arg is Supplier Supplier)
                    SupplierEntity = new SupplierEntity(Supplier);				
                if (arg is WareHouse WareHouse)
                    WareHouseEntity = new WareHouseEntity(WareHouse);				
                if (arg is ICollection<ReceiptNoteLine> ReceiptNoteLines)
                    ReceiptNoteLineEntities = ReceiptNoteLines.Select(model => new ReceiptNoteLineEntity(model, model.Product)).ToList();				
			}
        }
    }

    public class ReceiptNoteSearchEntity : FilterEntity
    {
        public Guid? Id { get; set; }
        public Int32? ReceiptNoteNo { get; set; }
        public DateTime? InvoiceDate { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public Boolean? Lock { get; set; }
        public String Title { get; set; }
        public DateTime? ReceiptDate { get; set; }
        public DateTime? SystemDate { get; set; }
        public String Address { get; set; }
        public String PhoneNumber { get; set; }
        public String Comment { get; set; }
        public Guid? WareHouseId { get; set; }
        public Guid? SupplierId { get; set; }
        public Guid? EmployeeId { get; set; }
    }
}
